cmd/snap-bootstrap/cmd_initramfs_mounts_test.go

```diff
@@ -932,6 +941,103 @@ func (s *initramfsMountsSuite) TestInitramfsMountsRunModeStep2KernelStatusTrying
 `, boot.InitramfsRunMntDir))
 }
 
+func (s *initramfsMountsSuite) TestUnlockIfEncrypted(c *C) {
+	for _, tc := range []struct {
+		hasTPM    bool
+		tpmErr    error
+		hasEncdev bool
+		last      bool
+		lockOk    bool
+		activated bool
+		device    string
+		err       string
+	}{
+		{true, nil, true, true, true, true, "name", ""},
```

Comment:
The test cases would be easier to follow if we used named fields here, and you only really need to specify the fields that need to have a non-default value:
```
{ 
    hasTPM: true, hasEncdev: true, last: true, lockOk: true, 
    activated: true, device: "name",
}, { 
    hasTPM: true, hasEncdev: true, last: true, lockOk: true, 
    err: "cannot activate encrypted device .*: activation error",
}, {
...
...
```